### PR TITLE
Kogito Configuration options for 9.2

### PR DESCRIPTION
**Pull request by LightGuard**
Adding generated configuration into the documentation

### CODE REVIEW

**Comment by LightGuard** [comment]
I recommend looking at the rendered diff and not the Asciidoc source itself. Tables can be difficult to read in Asciidoc

**Comment by LightGuard** [comment]
@gitgabrio mind taking a look and seeing if I have everything right?

File: reference-guide/configuration.adoc [hunk 1]
```diff
@@ -1,4 +1,372 @@
...
+h|[[kie-flyway_configuration]]link:#kie-flyway_configuration[Configuration property]
+h|Type
+h|Default
+
+a| `process`
+[.description]
+--
+String prefix for Kogito CloudEvents type fields.
+Since this is a required field, the constructor will fill them with this default value.
+Ideally, callers would use #TYPE_FORMAT to fill this field using the process name and the signal node name, e.g: process.travelagency.visaapproved
```
Comment (gitgabrio) [inline reference-guide/configuration.adoc:21]: Hi @pefernan could you check if this is an "application property" ?

File: reference-guide/configuration.adoc [hunk 2]
```diff
@@ -1,4 +1,372 @@
...
+--
+| string
+| 
+a| `/process/%s`
```
Comment (gitgabrio) [inline reference-guide/configuration.adoc:25]: Hi @pefernan same as above

File: reference-guide/configuration.adoc [hunk 3]
```diff
@@ -1,4 +1,372 @@
...
+[.description]
+--
+Environment variable injected by Knative
+--
+| string
+| 
+
+
+a| `MAIN`
+[.description]
```
Comment (gitgabrio) [inline reference-guide/configuration.adoc:196]: @LightGuard  to be removed - not an application property

File: reference-guide/configuration.adoc [hunk 4]
```diff
@@ -1,4 +1,372 @@
...
+--
+System property to override default output directory for generated PMML sources (inside *target*)
+--
+| string
+| 
+a| `generated-pmml-sources`
+[.description]
+--
+Default output directory for generated PMML sources (*target/generated-pmml-sources*)
```
Comment (gitgabrio) [inline reference-guide/configuration.adoc:367]: @LightGuard  to be removed - not an application property

File: reference-guide/configuration.adoc [hunk 5]
```diff
@@ -1,4 +1,372 @@
...
+| string
+| 
+a| `dumpKiePmmlSources`
+[.description]
+--
+System property to dump generated PMML sources <b>yes</b>
+--
+| string
+| 
+a| `dumpKiePmmlDirectory`
```
Comment (gitgabrio) [inline reference-guide/configuration.adoc:357]: @LightGuard  to be removed - not an application property

File: reference-guide/configuration.adoc [hunk 6]
```diff
@@ -1,4 +1,372 @@
...
+a| `/process/%s`
+[.description]
+--
+String format for Kogito CloudEvents source fields.
+Since this is a required field, the constructor will fill them with default value, e.g.: /process/travelagency
+See more about the source format: https://github.com/cloudevents/spec/blob/v1.0/spec.md#source-1
+--
+| string
+| 
+a| `http://drools.org/rules`
```
Comment (gitgabrio) [inline reference-guide/configuration.adoc:34]: @LightGuard  to be removed - not an application property

**Comment by LightGuard** [comment]
@pefernan please take a look so we can get this fixed or merged.

File: reference-guide/configuration.adoc [hunk 7]
```diff
@@ -1,4 +1,328 @@
...
 
-Please see the full https://docs.kogito.kie.org/latest/html_single/#chap-kogito-configuring[configuration reference] at the Kogito Documentation page.
\ No newline at end of file
+== {PRODUCT} Configuration Properties
+
+[.configuration-legend]
+icon:lock[title=Fixed at build time] Configuration property fixed at build time - All other configuration properties are overridable at runtime
+[.configuration-reference.searchable, cols="80,.^10,.^10"]
+|===
+h|[[kie-flyway_configuration]]link:#kie-flyway_configuration[Configuration property]
```
Comment (pefernan) [inline reference-guide/configuration.adoc:12]: @LightGuard this looks a bit weird, why we use kie-flyway_configuration in this header anchor? I'd say that the  ke-flyway config could go inside of this table. (kie.flyway.enabled: true|false)

File: reference-guide/configuration.adoc (hunk 1 above)
Comment (pefernan) [reply reference-guide/configuration.adoc:21]: @gitgabrio @LightGuard  this doesn't look like application property, this comes from the DataEvent metadata.

File: reference-guide/configuration.adoc (hunk 2 above)
Comment (pefernan) [reply reference-guide/configuration.adoc:25]: Same here

File: reference-guide/configuration.adoc [hunk 8]
```diff
@@ -1,4 +1,328 @@
...
+a| `/process/%s`
+[.description]
+--
+String format for Kogito CloudEvents source fields.
+Since this is a required field, the constructor will fill them with default value, e.g.: /process/travelagency
+See more about the source format: https://github.com/cloudevents/spec/blob/v1.0/spec.md#source-1
+--
+| string
+| 
+a| `openApiDescriptor`
```
Comment (pefernan) [inline reference-guide/configuration.adoc:34]: I'm not familiar with this one, but looking in the code I don't se it's being used.

File: reference-guide/configuration.adoc [hunk 9]
```diff
@@ -1,4 +1,328 @@
...
+--
+strategy for generating the configuration key of open API specifications; possible values are:
+file_name: uses the last element of the spec uri
+full_uri: uses the full path of the uri
+spec_title: uses the spec title
+function_name: uses the function name
+--
+| string
+| 
+a| `json`
```
Comment (pefernan) [inline reference-guide/configuration.adoc:72]: IDK what this is

File: reference-guide/configuration.adoc [hunk 10]
```diff
@@ -1,4 +1,328 @@
...
+[.description]
+--
+--
+| 
+| 
+a| `yml`
```
Comment (pefernan) [inline reference-guide/configuration.adoc:78]: same :S

Maybe related to swf format?

File: reference-guide/configuration.adoc [hunk 11]
```diff
@@ -1,4 +1,328 @@
...
+[.description]
+--
+--
+| 
+| 
+a| `transactionEnabled`
```
Comment (pefernan) [inline reference-guide/configuration.adoc:84]: looks like we are missing the `kogito.<generatorName>.` preffix here.

File: reference-guide/configuration.adoc [hunk 12]
```diff
@@ -1,4 +1,328 @@
...
+[.description]
+--
+Flag used to configure transaction enabling.
+--
+| boolean 
+| true 
+a| `filesystem`
```
Comment (pefernan) [inline reference-guide/configuration.adoc:91]: Looks like this one is a value for `kogito.persistence.type`, I think it shoudn't be here.

**Comment by LightGuard** [comment]
@gitgabrio and @pefernan Changes made per review comments. Please approve if it looks good and I'll get it squashed and merged.

**Pull request closed without merging**